sycl/doc/design/OffloadDesign.md

```diff
@@ -355,6 +403,33 @@ Additional options passed by the user via the
 `-Xsycl-target-backend=spir64_x86_64 <opts>` command will be processed by a new
 option to the wrapper, `--cpu-tool-arg=<arg>`
 
+Similar to SYCL offloading to Intel GPUs using `--offload-arch`, SYCL AOT for Intel CPUs
+will also leverage the `--offload-arch` option.
+The valid CPU device names accepted for `--offload-arch` are CPU names from ``clang -march``.
+These names are more verbose, and do not overlap with the Intel GPU names.
+These user input CPU names are mapped to the corresponding ``opencl-aot -march`` option.
+
+The following table shows a mapping of the accepted values for `--offload-arch` to enable SYCL offloading to Intel CPUs and the corresponding `-march` value passed to opencl-aot.
+
+| Intel CPU device | --offload-arch accepted value | opencl-aot -march value |
+|----------------|-------------------------|----------------------------|
+| Intel(R) Advanced Vector Extensions 512 | skylake-avx512 | avx512     |
+| Intel(R) Advanced Vector Extensions 2 | core-avx2 | avx2              |
+| Intel(R) Advanced Vector Extensions | corei7-avx | avx                |
+| Intel(R) Streaming SIMD Extensions 4.2 | corei7 | sse4.2              |
+| Intel(R) microarchitecture code name Westmere | westmere | wsm        |
+| Intel(R) microarchitecture code name Sandy Bridge | sandybridge | snb |
+| Intel(R) microarchitecture code name Ivy Bridge | ivybridge | ivyb    |
+| Intel(R) microarchitecture code name Broadwell | broadwell | bdw      |
+| Intel(R) microarchitecture code name Coffee Lake | coffeelake | cfl   |
+| Intel(R) microarchitecture code name Alder Lake | alderlake | adl     |
+| Intel(R) microarchitecture code name Skylake (server) | skx | skx     |
```

Comment:
Is there a reason for having `--offload-arch` accepted value for Skylake to be short and not as descriptive as the rest? Feels like we're not consistent with the rest.

Reply:
For SYCL offloading to Intel CPUs and GPUs, there is a mapping between the accepted values for `--offload-arch` and what is passed to the target-specific LLVM compiler (OCLOC or opencl-aot).
For NVidia GPUs and AMD GPUs, there is no such mapping. I can maybe add a list of accepted values for `--offload-arch `NVidia and AMD GPUs.

Reply:
I just noticed that Skylake client is missing from this list.
The `--offload-arch` accepted value for Skylake client is `skylake`, so `skx` was chosen for Skylake server.